src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+    threshold: 20,
+    units: 'px',
+    timeout: 500
+  };
+
+  protected constructor(public readonly target: Element, threshold?: string, timeout?: number) {
+    super();
+    this.config = this.getConfig(threshold, timeout);
+  }
+
+  /**
+   * Passes threshold into number and units, creates config from passed threshold and distance values or uses default ones.
+   * @param threshold - the distance threshold (px, vh, vw supported) that must be traveled before a swipe event is triggered.
+   * @param timeout - the time in milliseconds that must elapse between pointerdown and pointerup events before a wipe event is triggered.
+   * @returns ESLSwipeEventTarget configuration {@link SwipeEventTargetConfig}.
+   */
+  protected getConfig(threshold?: string, timeout?: number): SwipeEventTargetConfig {
+    const config = ESLSwipeEventTarget.defaultConfig;
+
+    if (threshold) {
+      ESLSwipeEventTarget.unitsAvailable.forEach((unit: string) => {
+        if (threshold.indexOf(unit) > 0) {
+          config.units = unit;
+          config.threshold = parseInt(threshold.replace(unit, ''), 10) || ESLSwipeEventTarget.defaultConfig.threshold;
+        }
+      });
+    }
+
+    config.timeout = timeout || config.timeout;
+
+    return config;
+  }
+
+  /**
+   * @param $el - the element to listen for swipe events on
+   * @param threshold - the distance threshold (px, vh, vw supported) that must be traveled before a swipe event is triggered. Optional. Defaults to 20px
+   * @param timeout - the time in milliseconds that must elapse between pointerdown and pointerup events before a wipe event is triggered. Optional.
+   * Defaults to 500.
+   * @returns Returns the instance of ESLSwipeEventTarget {@link ESLSwipeEventTarget}.
+   */
+  public static for($el: Element | ESLMixinElement, threshold?: string, timeout?: number): ESLSwipeEventTarget {
+    if ($el instanceof ESLMixinElement) return ESLSwipeEventTarget.for($el.$host, threshold, timeout);
+
+    return new ESLSwipeEventTarget($el, threshold, timeout);
+  }
+
+  /**
+   * Subscribes to pointerup and pointerdown event
+   */
+  public override addEventListener(callback: EventListener): void;
+  public override addEventListener(event: SwipeEventName, callback: EventListener): void;
+  public override addEventListener(event: any, callback: EventListener = event): void {
+    super.addEventListener(event, callback);
+    if (this.getEventListeners().length > 1) return;
+
+    this.target.addEventListener('pointerdown', this.handleStart.bind(this), false);
+    this.target.addEventListener('pointerup', this.handleEnd.bind(this), false);
+  }
+
+  /**
+   * Saves swipe start event target, time when swipe started, pointerdown event and coordinates.
+   * @param e - pointer event
+   */
+  protected handleStart(e: PointerEvent): void {
+    this.startEl = e.target as HTMLElement;
+    this.timeDown = e.timeStamp;
+    this.startEvent = e;
+    this.xDown = e.clientX;
+    this.yDown = e.clientY;
```

Comment:
We already have an event, do we need to extract point coordinates and timestamps here?

Reply:
It's used to calculate distance and time between start and end points and based on config make a decision weather to trigger swipe event or not

Reply:
Yes that's clear, but we have the `pointdown` event stored and available in `pointerup` handler
Event is finalized object, so `e.clientX` will not change in `pointerup` handler.